Source/Request.swift

```diff
@@ -605,7 +610,7 @@ extension Request: CustomDebugStringConvertible {
 public protocol RequestDelegate: AnyObject {
     var sessionConfiguration: URLSessionConfiguration { get }
 
-    func willRetryRequest(_ request: Request) -> Bool
+    func willAttemptToRetryRequest(_ request: Request) -> Bool
```

Comment:
Refactored this `RequestDelegate` API because it wasn't accurately named before. Previously, it was saying that if `true` was returned, the request would be retried. However, the existence of a retrier doesn't mean it will necessarily be retried, just that the retrier will inspect it to determine whether it should be retried or not.

Because of this, I refactored it to `willAttemptToRetry` instead. The implementation now also considers the existence of retry policies.